testapi.pm

```diff
@@ -962,23 +963,20 @@ sub script_run {    # no:style:signatures
         }, ['timeout'], @_);
 
     bmwqemu::log_call(cmd => $cmd, %args);
-    my $die_on_timeout = delete $args{die_on_timeout};
+    my $die_on_timeout = delete $args{die_on_timeout} // 1;
     my $ret = $distri->script_run($cmd, %args);
     if ($args{timeout} > 0) {
-        if ($die_on_timeout > 0) {
-            croak("command '$cmd' timed out") if !defined($ret);
+        if ($die_on_timeout == 0) {
+            # This is to warn users of deprecated behaviour of script_run()
+            my ($package, $filename, $line) = caller;
+            my $casedir = testapi::get_var(CASEDIR => '');
+            $filename =~ s%^\Q$casedir\E/%%;
+            bmwqemu::fctwarn("DEPRECATED call of script_run() in $filename:$line " .
+                  'requested by `die_on_timeout => 0` or set
+                  $distri->{script_run_die_on_timeout}. Adapt the test code to work
+                  with the default. This workaround will be removed in the near future');
```

Comment:
```suggestion
                  with the default. This workaround is going to be removed in the near future.');
```
I guess it doesn't really matter much, though - and one generally uses the will future to make threats :-)

Reply:
yes. I wanted to make that threat :) I think the message sounds more likely to happen with "will be removed" so that people would really act. I would prefer to keep that.